src/Engine/Graphics/RenderBase.cpp

```diff
@@ -118,7 +125,10 @@ unsigned int RenderBase::Billboard_ProbablyAddToListAndSortByZOrder(float z) {
 void RenderBase::DrawSpriteObjects() {
     for (unsigned int i = 0; i < pSpriteObjects.size(); ++i) {
         // exit if we are at max sprites
-        if (::uNumBillboardsToDraw >= 500) break;
+        if (::uNumBillboardsToDraw >= 500) {
```

Comment:
By the way this is pretty legacy (you had what, 32Mb RAM back then?) don't see any sense in imposing any of these limits, or maybe make them configurable (e.g for android maybe?)

Reply:
Yeah, mm7 had 16Mb heap limit in D3D3 and 27Mb in software. Though memory is not an issue even on phones. Main issue is that we still single-threaded for most part and arm cores on mobiles are still pretty weak to run single-threaded. And GPU's horsepower is also miles behind desktop beasts...

Reply:
Better configuration for graphics options is on the Todo list. It's still quite rare to bump up against this limit though.

Reply:
yeah we can absolutely leave it for later